### PR TITLE
Have rkt implement the container Runtime interface

### PR DESCRIPTION
**Pull request by vmarmol**
/cc @yujuhong @yifan-gu @dchen1107

### CODE REVIEW

**Comment by jonboulle** [comment]
lgtm

File: pkg/kubelet/container/runtime.go [hunk 1]
```diff
@@ -69,11 +69,12 @@ type Runtime interface {
 	ListImages() ([]Image, error)
 	// Removes the specified image.
 	RemoveImage(image string) error
+	// TODO(vmarmol): Unify pod and containerID args.
 	// GetContainerLogs returns logs of a specific container. By
 	// default, it returns a snapshot of the container log. Set 'follow' to true to
 	// stream the log. Set 'follow' to false and specify the number of lines (e.g.
 	// "100" or "all") to tail the log.
-	GetContainerLogs(containerID, tail string, follow bool, stdout, stderr io.Writer) (err error)
+	GetContainerLogs(podUID *api.Pod, containerID, tail string, follow bool, stdout, stderr io.Writer) (err error)
```
Comment (yujuhong) [inline pkg/kubelet/container/runtime.go]: s/podUID/pod

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1386,7 +1386,11 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName, tail stri
 		// waiting state.
 		return err
 	}
-	return kl.containerManager.GetContainerLogs(containerID, tail, follow, stdout, stderr)
+	pod, ok := kl.GetPodByFullName(podFullName)
```
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: Is it still necessary to get the pod status and perform various checks above? It feels like those could be redundant after your change.

**Comment by yujuhong** [comment]
Looks good overall with one nit and a question :)

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: I'd be fine with adding a TODO to re-evaluate and remove the redundancy later, if it's preferred :)

Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: We do today to get the containerID, I think we may be able to remove that need with using the container name. I'll add a TODO, it's not super clear to me what to do as of yet though.

File: pkg/kubelet/container/runtime.go (hunk 1 above)
Comment (vmarmol) [inline pkg/kubelet/container/runtime.go]: Done

**Comment by vmarmol** [comment]
Thanks for the review @yujuhong! PTAL

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: IMO, GetContainerLogs should simply find the containerID itself and return error if it's not found. Also, not relevant to this PR, but why can't GetContainerLogs take containerName as an argument -- this seems to fit well with our pod-level abstraction.

**Comment by yujuhong** [comment]
LGTM, but you need a rebase ;)

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: +1 that's what the interface should be.

**Comment by vmarmol** [comment]
Ah thanks I didn't notice :) rebased.

**Comment by yujuhong** [comment]
LGTM. Will merge on green!

**Comment by yujuhong** [comment]
Shippable passed. Merging.

**Pull request merged**